runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -285,16 +286,26 @@ jvmtiGetFrameCount(jvmtiEnv* env,
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE, &isVirtual);
 		if (rc == JVMTI_ERROR_NONE) {
 			J9StackWalkState walkState;
-
-			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-
-			walkState.walkThread = targetThread;
 			walkState.flags = J9_STACKWALK_INCLUDE_NATIVES | J9_STACKWALK_VISIBLE_ONLY;
 			walkState.skipCount = 0;
-			vm->walkStackFrames(currentThread, &walkState);
+
+			BOOLEAN callResume = TRUE;
+			if ((NULL == targetThread) && isVirtual) {
+				j9object_t threadObject = J9_JNI_UNWRAP_REFERENCE(thread);
+				j9object_t contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
+				walkContinuationStackFrames(currentThread, contObject, &walkState);
+				callResume = FALSE;
+			} else {
+				vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
+				walkState.walkThread = targetThread;
+				vm->walkStackFrames(currentThread, &walkState);
+			}
+
 			rv_count = (jint) walkState.framesWalked;
 
-			vm->internalVMFunctions->resumeThreadForInspection(currentThread, targetThread);
+			if (callResume) {
+				vm->internalVMFunctions->resumeThreadForInspection(currentThread, targetThread);
+			}
```

Comment:
no need to use `callResume`, you can just move the resume into the else block after `vm->walkStackFrames`